app/routes/user/email/index.tsx

```diff
@@ -7,53 +7,79 @@
  */
 import type { DataFunctionArgs } from '@remix-run/node'
 import { redirect } from '@remix-run/node'
-import { useLoaderData } from '@remix-run/react'
+import { useFetcher, useLoaderData } from '@remix-run/react'
+import { Button, ButtonGroup, Grid, Icon } from '@trussworks/react-uswds'
+import { useState } from 'react'
 
-import { EmailNotificationServer } from '../email_notifications.server'
+import {
+  createGcnCircularEmailNotification,
+  deleteGcnCircularEmailNotification,
+  getUsersCircularSubmissionStatus,
+} from '../email_circulars.server'
+import {
+  deleteEmailNotification,
+  getEmailNotifications,
+  sendTestEmail,
+} from '../email_notices.server'
 import EmailNotificationCard from '~/components/EmailNotificationCard'
 import HeadingWithAddButton from '~/components/HeadingWithAddButton'
 import SegmentedCards from '~/components/SegmentedCards'
+import Spinner from '~/components/Spinner'
+import { getHostname } from '~/lib/env.server'
 import { getFormDataString } from '~/lib/utils'
+import { getUser } from '~/routes/__auth/user.server'
 
 export const handle = { getSitemapEntries: () => null }
 
 export async function action({ request }: DataFunctionArgs) {
+  const user = await getUser(request)
+  if (!user) throw new Response(null, { status: 403 })
   const data = await request.formData()
   const uuid = getFormDataString(data, 'uuid')
   const intent = getFormDataString(data, 'intent')
+
   switch (intent) {
     case 'delete':
       if (uuid) {
-        const machine = await EmailNotificationServer.create(request)
-        await machine.deleteEmailNotification(uuid)
+        await deleteEmailNotification(uuid, user.sub)
       }
     case 'sendTest':
       const recipient = getFormDataString(data, 'recipient')
       if (recipient) {
-        const machine = await EmailNotificationServer.create(request)
-        await machine.sendTestEmail(recipient)
+        await sendTestEmail(recipient)
       }
+    case 'subscribe':
+      await createGcnCircularEmailNotification(user?.sub, user?.email)
```

Comment:
Did TypeScript not infer that user must be defined from the `if (!user) throw ...` above?

```suggestion
      await createGcnCircularEmailNotification(user.sub, user.email)
```